### PR TITLE
fix(javascript): use requirejs where needed

### PR DESCRIPTION
**Pull request by lepokle**
RequireJS and loading via script tags conflicts, if script tags are behind the
require definition. This commit converts some old code to use require to minimize conflicts.
However there are a lot of files left that have to be migrated.

Closes sw360/sw360portal#598
Closes sw360/sw360portal#594
Closes sw360/sw360portal#592
Closes sw360/sw360portal#607

### CODE REVIEW

**Comment by mcjaeger** [comment]
tested with firefox :
#594 
#592 
seem to be OK, but 
* Edit button at release detail view does not work
* Edit button for components and projects works
* Edit in components and then jumping to the release works also to edit a release

seen error in firefox:
```
ReferenceError: event is not defined[Learn More]  components:116:103
	<anonymous> https://localhost:8443/group/guest/components:116:103
	dispatch jquery.js:5225:15
	add/r.handle jquery.js:4878:5

        
```

**Comment by lepokle** [comment]
@mcjaeger thanks for testing. Found out a funny difference between firefox and chrome due to this bug. Chrome defines a global event object for event listeners. Therefore the bug did not come up in Chrome. See also: https://stackoverflow.com/questions/40871422/chrome-seems-to-define-event-variable-in-meteor-blaze

Fixed.

**Comment by mcjaeger** [comment]
@adrian-evo if you would have mind, mind a retest?

**Comment by lepokle** [comment]
Just checked for onclick handlers again. None critical found, all remaining onclick-handlers should work.
However, found a non-working datapicker -> fixed.

File: frontend/sw360-portlet/src/main/webapp/js/utils/includes/fossologyClearing.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        });
+    }
+
+    return {
+        initialize: initialize,
+        openSelectClearingDialog: function(prefix, releaseId) {
+            openSelectClearingDialog(prefix + releaseId, releaseId);
+        }
+    }
+});
```
Comment (alexbrdn) [inline frontend/sw360-portlet/src/main/webapp/js/utils/includes/fossologyClearing.js]: missing newline at EOF

File: frontend/sw360-portlet/src/main/webapp/js/modules/tabview.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+            options.srcNode = '#' + id;
+
+            tabView = new Y.TabView(options).render();
+        });
+    }
+
+    return {
+        create: createTabView
+    };
+});
```
Comment (alexbrdn) [inline frontend/sw360-portlet/src/main/webapp/js/modules/tabview.js]: missing newline at EOF

File: frontend/sw360-portlet/src/main/webapp/html/components/includes/releases/clearingDetails.jspf [hunk 3]
```diff
@@ -157,4 +157,14 @@
...
+    require(['jquery', 'utils/includes/fossologyClearing'], function ($, fossology) {
+        $(window).load(function () {
+            fossology.initialize();
+        });
+        $('#ReleaseClearingOverview').on('click', '.action.send-to-fossology', function(event) {
+            var data = $(event.currentTarget).data();
+            fossology.openSelectClearingDialog('fossologySending', data.releaseId);
+        });
+    });
+</script>
```
Comment (alexbrdn) [inline frontend/sw360-portlet/src/main/webapp/html/components/includes/releases/clearingDetails.jspf]: missing newline at EOF

**Comment by adrian-evo** [comment]
Works as expected. Additionally also #607 not reproducible on this branch.

**Comment by mcjaeger** [comment]
@adrian-evo thank you very much, this is very good news.

**Pull request merged**